### PR TITLE
sol

### PR DESCRIPTION
**Pull request by vladyslavchichkan**

### CODE REVIEW

**Review by Arsen-hrynevych — CHANGES_REQUESTED** [review]
Пофікси помилки з тестами.

File: cinema/models.py [hunk 1]
```diff
@@ -1,9 +1,26 @@
...
+class CinemaHall(models.Model):
+    name = models.CharField(max_length=255)
+    rows = models.IntegerField()
+    seats_in_row = models.IntegerField()
+
+
 class Movie(models.Model):
     title = models.CharField(max_length=255)
     description = models.TextField()
+    actors = models.ManyToManyField(Actor)
```
Comment (maxgerman) [inline cinema/models.py]: what if we want to get the movies from the actor (reverse relationship)?

File: cinema/views.py [hunk 2]
```diff
@@ -1,45 +1,97 @@
...
             return Response(serializer.data, status=status.HTTP_201_CREATED)
-
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
 
-@api_view(["GET", "PUT", "DELETE"])
-def movie_detail(request, pk):
-    movie = get_object_or_404(Movie, pk=pk)
+class GenreDetail(APIView):
+    @staticmethod
```
Comment (maxgerman) [inline cinema/views.py]: why staticmethod?

File: cinema/views.py [hunk 3]
```diff
@@ -1,45 +1,97 @@
...
+        serializer = GenreSerializer(self.get_object(pk=pk), data=request.data)
         if serializer.is_valid():
             serializer.save()
             return Response(serializer.data, status=status.HTTP_200_OK)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
+    def patch(self, request, pk):
+        serializer = GenreSerializer(self.get_object(pk=pk),
+                                     data=request.data, partial=True)
+        if serializer.is_valid():
```
Comment (maxgerman) [inline cinema/views.py]: how can we avoid this "if" and save the same functionality?

**Review by maxgerman — CHANGES_REQUESTED** [review]
Good job!
See comments

File: cinema/views.py (hunk 3 above)
Comment (vladyslavchichkan) [reply cinema/views.py]: we can use "try"

File: cinema/models.py (hunk 1 above)
Comment (vladyslavchichkan) [reply cinema/models.py]: we can use related_name

File: cinema/views.py (hunk 2 above)
Comment (vladyslavchichkan) [reply cinema/views.py]: pycharm reccomends to use static here, all the same if we use it or not

**Review by Arsen-hrynevych — CHANGES_REQUESTED** [review]
Unresolved comments from the previous mentor, and I see the possibility that you could format the code for better readability.

**Review by Arsen-hrynevych — APPROVED** [review]
Thanks. Try to find a shorter way to write the same functionality.

**Pull request closed without merging**